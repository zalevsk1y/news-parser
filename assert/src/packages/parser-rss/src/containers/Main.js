import React from 'react';
import Message from '@news-parser/message/';
import InputForm from './InputForm';
import Posts from './Posts';
import Indicator from './Indicator';
import {connect} from 'react-redux';
import {fetchList} from '../actions/list.actions';
import VisualConstructor from '@news-parser/visual-constructor/';
import PropTypes from 'prop-types';
import Translate from './Translate';
/**
 * Main application element.
 * 
 * @since 0.8.0
 */

class Main extends React.Component {
    constructor(props){
        super(props);
   
    }
  
    render() {
        return (
            <div className={"wrap wrap-parsing"} >
                <VisualConstructor />
                <div className="parsing-title">
                    <h1><Translate>News-Parser</Translate></h1>
                </div>
                <Indicator step={0.5}/>
                <Message />
                <InputForm />
                <Posts />
            </div>
        )
    }
}



export default connect()(Main);


Main.propTypes={
    /**
     * Current app parsing entity [list|single|multi]
     */
    entity:PropTypes.string,
    /**
     * Current event, that apply on current app entity [dialog]
     */
    event:PropTypes.string,
    /**
     * RSS data url.
     */
    url:PropTypes.string,
    /**
     * Dialog window state.
     */
    dialog:PropTypes.oneOfType([
        PropTypes.object,
        PropTypes.bool
    ]),

  
}