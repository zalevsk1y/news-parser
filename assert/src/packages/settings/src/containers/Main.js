import React from 'react';
import Tab from '../components/Tab';
import Button from '../components/Button';
import GeneralPage from './GeneralPage';
import PostPage from './PostPage';
import GalleryPage from './GalleryPage'
import {connect} from 'react-redux'
import {main} from '../actions/index'
import {getNonce} from '@news-parser/helpers';
import Message from "./Message";
import PropTypes from 'prop-types';
import Translate from './Translate';

export class Main extends React.Component {
    constructor(props){
      super(props);

      this.state={tabsArray:props.tabs};
      this.createTabs=this.createTabs.bind(this)
      this.selectHandler=this.selectHandler.bind(this)
      this.updateHandler=this.updateHandler.bind(this);
      this.saveSettings=this.saveSettings.bind(this);
      this.resetToDefault=this.resetToDefault.bind(this)
    }
    createTabs(args){
      return args.tabs.map(function(item,key){
        return (<Tab className={item.className} active={item.active} key={key} id={key} selectHandler={this.selectHandler} name={item.name} />);
      }.bind(this));
  
    }
    selectHandler(id){
      const state= this.state.tabsArray.map(function(item,key){
        item.active=key===id?true:false;
        return item;
      });
      this.setState({tabsArray:state,currentID:id});
    }
    saveSettings(){
      const nonce=getNonce({page:'settings',action:'save'})
      this.props.saveSettings(this.props.settings,nonce)
    }
    resetToDefault(){
      const nonce=getNonce({page:'settings',action:'get'})
      this.props.resetToDefault(nonce)
    }
    updateHandler(parameterName,state){
      const settings={...this.state.settings};
      settings[parameterName]=state;
      this.setState({settings:settings})
    }
    renderPage(props){
   
      switch(props.name){
        case 'General':
          return (<GeneralPage  />);
        case "Post":
          return (<PostPage  />);
        case "Gallery":
          return (<GalleryPage  />);
        default:
          return (<GeneralPage  />);
      }
      
     
    }
    translate(text){
      return (<Translate>{text}</Translate>)
    }
    render() {
      var tabId=this.state.currentID||0;
      return (
        <div className='wrap wrap-parsing'>
          <Message />
          <div className="parsing-title">
              <h1>
                <Translate>Settings</Translate>
                </h1>
          </div> {/*parsing-title*/}
          <div className="settings-main-tabs" >
             <this.createTabs tabs={this.state.tabsArray}/>
          </div> {/*settings-main-tabs*/}
          <div className='settings-container'>
            <this.renderPage name={this.state.tabsArray[tabId].name}/>
          </div>{/*settings-main-tabs*/}
          <Button onClick={this.saveSettings} value={this.translate("Save settings")} className='button-primary' action='save'></Button>
          <Button onClick={this.resetToDefault} value={this.translate('Reset settings')} className='button-secondary' action='reset'></Button>
        </div> 
      );
    }
  }
  function mapStateToProps(state){
      return {
        settings:state.settings
      }
  }
  function mapDispatchToProps (dispatch){
    
   
    return {
  
      resetToDefault:(nonce)=>{
        dispatch(main.getDefaultSettingsFromServer(dispatch,nonce))
      },
      saveSettings:(settings,nonce)=>{
        dispatch(main.saveSettingsToServer(dispatch,settings,nonce));
      }

    }
  }
  export default connect(mapStateToProps,mapDispatchToProps)(Main);

  Main.propTypes={
    tabs:PropTypes.array.isRequired,
    settings:PropTypes.object.isRequired,

    resetToDefault:PropTypes.func.isRequired,
    saveSettings:PropTypes.func.isRequired
  }