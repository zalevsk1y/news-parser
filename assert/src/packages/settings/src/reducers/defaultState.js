
export const defaultState={ 
    main:{
        isFetching:false,
        error:false,
        message:false,
    },
    settings:{
        general:{
            addSource:false
        },
        post:{
            addThumbnail:true,
            parseOtherPictures:false,
            showPicturesDialog:true,
            maxPictures:5,
        },
        gallery:{
            addGallery:false,
            shortCode:undefined,
            parameterName:undefined
        }
}
};
