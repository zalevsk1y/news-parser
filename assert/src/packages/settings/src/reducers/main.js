
import {types} from '../actions/index';


export function main(state={},action){
    switch (action.type){
        case types.main.SEND_REQUEST:
            return {...state,
                isFetching:true};
        case types.main.RECEIVE_RESPONSE:
            return {...state,
                isFetching:false,
                error:action.data.err,
                message:action.data.msg,
                lang:action.data.lang
            }
        case types.main.FETCH_ERROR:
            return {...state,
                isFetching:false,
                error:action.data.error,
                message:action.data.message
            }
        default:
            return {...state,
                isFetching:false,
                error:false,
                message:false,
              
    };
    }
}

