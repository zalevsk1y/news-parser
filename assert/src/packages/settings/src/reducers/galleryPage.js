
import {types} from '../actions/index';
import { defaultState } from './defaultState';


const galleryPage=(state=defaultState.settings.gallery,action)=>{
 
    switch (action.type){
        case types.main.SET_ALL_SETTINGS:
            return {...action.settings.gallery};
        case types.gallery.TOGGLE_ADD_GALLERY:
            return{...state,
            addGallery:!state.addGallery
        };
        case types.gallery.CHANGE_SHORTCODE:
            return {...state,
            shortCode:action.text
        };
        case types.gallery.CHANGE_PARAMETER_NAME:
            return {...state,
            parameterName:action.text
        };
        default:
        return {...state}
    }
}
export default galleryPage;