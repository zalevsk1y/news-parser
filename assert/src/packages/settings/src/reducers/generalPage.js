
import {types} from '../actions/index';
import { defaultState } from './defaultState';

const generalPage=(state=defaultState.settings.general,action)=>{
    switch(action.type){
        case types.main.SET_ALL_SETTINGS:
            return {...action.settings.general}
        case types.general.TOGGLE_ADDSOURCE:
            return{...state,
                'addSource':!state.addSource
            }
        default:
            return{
                ...state
            }
    }
    }

export default generalPage;