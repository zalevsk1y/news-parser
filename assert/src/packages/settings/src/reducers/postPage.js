
import {types} from '../actions/index';
import { defaultState } from './defaultState';

const postPage=(state=defaultState.settings.post,action)=>{
    switch (action.type){
        case types.main.SET_ALL_SETTINGS:
            return {...action.settings.post}
        case types.post.TOGGLE_ADD_THUMBNAIL:
            return {...state,addThumbnail:!state.addThumbnail}
        case types.post.TOGGLE_PARSE_OTHER_PICTURES:
            return {...state,parseOtherPictures:!state.parseOtherPictures};
        case types.post.TOGGLE_SHOW_PICTURES_DIALOG:
            return {...state,showPicturesDialog:!state.showPicturesDialog};    
        case types.post.CHANGE_MAX_PICTURES:
            return {...state,maxPictures:action.number}
        default:
            return state;
    }
}
export default postPage