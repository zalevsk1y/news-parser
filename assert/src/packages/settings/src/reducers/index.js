import general from './generalPage.js';
import post from './postPage.js';
import gallery from './galleryPage.js';
import {main} from './main.js'
import {combineReducers} from 'redux';

const settings=combineReducers({
    general,
    post,
    gallery,
})
const  settingsReducer=combineReducers({
    main,
    settings
})

function flowLeftToRight(funcObject){
    return (state,action)=>{
        let newState=state;
        const funcArray=Object.keys(funcObject)
        for(var i=0,end=funcArray.length;i<end;i++){
            var funcName=funcArray[i];
            newState=funcObject[funcName].apply(this,[newState,action]);
        }
        return newState;
    }
}
export default settingsReducer;