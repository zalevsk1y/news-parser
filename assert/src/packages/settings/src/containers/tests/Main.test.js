import React from 'react';
import {shallow,configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {Main} from '../Main';
import {defaultState} from '../../reducers/defaultState'


configure({adapter:new Adapter()});
describe('Main',()=>{
    it('should render correctly',()=>{
        const dispatchToProps={
            getSettings:jest.fn(),
            resetToDefault:jest.fn(),
            saveSettings:jest.fn()
        };
        const component=shallow( <Main
            settings={defaultState.settings}
            tabs={[{   
            className:'general',
            status:true,
            name:'General'
        },
        {   
            className:'post',
            status:false,
            name:'Post'
        }]}  {...dispatchToProps}/>)
        expect(component).toMatchSnapshot();
    });
    
})