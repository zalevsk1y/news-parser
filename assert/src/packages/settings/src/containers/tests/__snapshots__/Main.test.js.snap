// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main should render correctly 1`] = `
<div
  className="wrap wrap-parsing"
>
  <Connect(Message) />
  <div
    className="parsing-title"
  >
    <h1>
      <Connect(Translate)>
        Settings
      </Connect(Translate)>
    </h1>
  </div>
   
  <div
    className="settings-main-tabs"
  >
    <bound createTabs
      tabs={
        Array [
          Object {
            "className": "general",
            "name": "General",
            "status": true,
          },
          Object {
            "className": "post",
            "name": "Post",
            "status": false,
          },
        ]
      }
    />
  </div>
   
  <div
    className="settings-container"
  >
    <renderPage
      name="General"
    />
  </div>
  <Button
    action="save"
    className="button-primary"
    onClick={[Function]}
    value={
      <Connect(Translate)>
        Save settings
      </Connect(Translate)>
    }
  />
  <Button
    action="reset"
    className="button-secondary"
    onClick={[Function]}
    value={
      <Connect(Translate)>
        Reset settings
      </Connect(Translate)>
    }
  />
</div>
`;
